pkg/network/event_common_linux.go

```diff
@@ -0,0 +1,47 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build linux
+// +build linux
+
+package network
+
+// Sub returns s-other
+func (s StatCounters) Sub(other StatCounters) (sc StatCounters, underflow bool) {
+	if s.Retransmits < other.Retransmits && s.Retransmits > 0 ||
+		(s.TCPClosed < other.TCPClosed && s.TCPClosed > 0) ||
+		(s.TCPEstablished < other.TCPEstablished && s.TCPEstablished > 0) ||
+		isUnderflow(other.RecvBytes, s.RecvBytes, maxByteCountChange) ||
+		isUnderflow(other.SentBytes, s.SentBytes, maxByteCountChange) {
+		return sc, true
+	}
+
+	sc = StatCounters{
+		RecvBytes: s.RecvBytes - other.RecvBytes,
+		SentBytes: s.SentBytes - other.SentBytes,
+	}
+
+	// on linux, sent and recv packets are collected
+	// as uint32's, but StatCounters stores them as uint64,
+	// so we need to treat them as uint32 to detect underflows
+	sc.RecvPackets = uint64(uint32(s.RecvPackets) - uint32(other.RecvPackets))
+	sc.SentPackets = uint64(uint32(s.SentPackets) - uint32(other.SentPackets))
+	if (s.RecvPackets < other.RecvPackets && sc.RecvPackets > maxPacketCountChange) ||
+		(s.SentPackets < other.SentPackets && sc.SentPackets > maxPacketCountChange) {
```

Comment:
Seems odd to have an isUnderflow for the previous if, but to do it manually for this one

Reply:
Special case for the packet counts, since we need to do the subtraction differently. `isUnderflow` for not get the delta right by just subtracting `uint64`'s.